### ISSUE TITLE
multiple outputs

### ISSUE THREAD

**Issue by Ivashar** [microsoft/CNTK#3759]
Hi, I created a network with several outputs, for each output I created my own loss function, but when I call the CreateTrainer () function I get an exception:	

**About to throw exception 'A Function instance 'Composite(Combine): Input('features0', [32], [0, #]), Input('labels0', [2
], [#]), Input('features1', [32], [1, #]), Input('labels1', [2], [#]) -> Output('Block733_Output_0', [1], [#]), Output('
Block776_Output_0', [1], [#])' with more than one output cannot be implicitly converted to a Variable.'**

My code:

vector<Variable> labels;
	labels.push_back(InputVariable({ numOutputClasses }, false, DataType::Float, L"labels0", {Axis::DefaultBatchAxis()}));
	trainingLoss = SquaredError(lstmModel->Outputs()[0],  labels.at(0) );
	prediction = ClassificationError(lstmModel->Outputs()[0], labels.at(0));

	for (int k = 1; k < features.size(); ++k)
	{
		labels.push_back(InputVariable({ numOutputClasses }, false, DataType::Float, L"labels" + to_wstring(k), { Axis::DefaultBatchAxis() }));
		
		trainingLoss = Combine({ trainingLoss , SquaredError(lstmModel->Outputs()[k], labels.at(k)) });
		prediction = Combine({ prediction , ClassificationError(lstmModel->Outputs()[k], labels.at(k)) });
	}

	//create learner 
	paramLearner = AdamLearner(lstmModel->Parameters(), 
									learningRate, momentumSchedule, DefaultUnitGainValue, DefaultVarianceMomentum);
	
	//paramLearner = MomentumSGDLearner(lstmModel->Parameters(), learningRate, momentumSchedule);
	FunctionPtr comOut = Combine({ lstmModel->Outputs()[0], lstmModel->Outputs()[1] });

	//create trainer
	trainer = CreateTrainer(comOut, { trainingLoss }, prediction, vector<LearnerPtr>({ paramLearner }));   // HERE EXCEPTION

**Comment by delzac** [comment]
You can only have one final loss function. Optimising multiple loss function at the same time is an open area of research.

What you can do now is to combine multiple loss function into one like the following:
```
    loss1 = L1(network_output1, truth1)
    loss2 = L2(network_output2, truth2)
    loss = 0.5 * (loss1 + loss2)
```

**Comment by Ivashar** [comment]
delzac, thanks for the answer and apologize for my english. Are you sure what you are saying? I solved the problem and it works! It was necessary to use the operator overload "+" to combine the loss functions, and not the Combine () function. Why should there be a problem of simultaneous operation of the loss function? In fact, there are several networks operating in parallel, each with its own loss function. The result of each is simply averaged, as you showed in your example, I needed it, but what you thought, assuming my goal, is crazy!

**Comment by delzac** [comment]
So firstly, you can only have one loss function. It doesn't matter if you are doing deep learning or not. There are currently no optimisation algorithm that can work on multiple loss function at the same time.

If you have several network operating in parallel and do not interact with each other, then you should have a separate Trainer for each of them, with each trainer targeting a different GPU.

**Comment by Ivashar** [comment]
What did I do then and does it work?)

**Comment by delzac** [comment]
What do you mean?

Separate models should be trained separately.

If you have one model that used several loss function then the multiple loss functions need to be mathematically combined as you saw in my example where `loss = loss1 + loss2`

**Comment by Ivashar** [comment]
We are talking about the same thing, it's all my English.

	for (int k = 1; k < features.size(); ++k)
	{
		labels.push_back(InputVariable({ numOutputClasses }, false, DataType::Float, L"labels" + to_wstring(k), { Axis::DefaultBatchAxis() }));
		
		trainingLoss = trainingLoss + SquaredError(lstmModel->Outputs()[k], labels.at(k));
		prediction =  prediction + ClassificationError(lstmModel->Outputs()[k], labels.at(k));
	}


I want to learn tensorflow 2.0 python API. I don’t know where to start) Does CNTK and TF use Keras API? Therefore, knowing the Keras API, can you use TF freely?

**Comment by delzac** [comment]
Keras does offer CNTK and TK as backend. But moving forward Keras will only support TF in the newest release.

Keras API is in python.
